llvm/lib/CodeGen/SelectionDAG/SelectionDAGISel.cpp

```diff
@@ -2273,11 +2273,24 @@ void SelectionDAGISel::Select_PATCHPOINT(SDNode *N) {
   Ops.push_back(NumArgs);
 
   // Calling convention.
+  bool IsAnyRegCC = cast<ConstantSDNode>(It->getNode())->getZExtValue() ==
+                    CallingConv::AnyReg;
   Ops.push_back(*It++);
 
-  // Push the args for the call.
-  for (uint64_t I = cast<ConstantSDNode>(NumArgs)->getZExtValue(); I != 0; I--)
-    Ops.push_back(*It++);
+  uint64_t ExpectArgs = cast<ConstantSDNode>(NumArgs)->getZExtValue();
+  if (IsAnyRegCC) {
+    // Push the args for the call by scanning for `NextLive` markers.
+    while (ExpectArgs) {
+      SDNode *ItN = It->getNode();
+      if ((ItN->getOpcode() == ISD::TargetConstant) &&
+          (cast<ConstantSDNode>(ItN)->getZExtValue() == StackMaps::NextLive))
```

Comment:
I misread this one at first: it really feels like `{...}` brackets would help here?

Reply:
Fixed in [362242d](https://github.com/ykjit/ykllvm/pull/40/commits/362242df9a8d6176925ec958c1c9b92abc479103)